draft-ietf-dnsop-svcb-https.md

```diff
@@ -867,7 +867,32 @@ compliant recursive resolvers.  When SvcDomainName is ".", server operators
 SHOULD NOT include these hints, because they are unlikely to convey any
 performance benefit.
 
+## "mandatory" {#mandatory}
 
+In a ServiceForm RR, a SvcParamKey is considered "mandatory" if the RR will not
+function correctly for clients that ignore this SvcParamKey.  Each SVCB
+protocol mapping SHOULD specify a set of keys that are "automatically
+mandatory", i.e. mandatory if they are present in an RR.  The SvcParamKey
+"mandatory" is used to indicate any mandatory keys for this RR, in addition to
+any automatically mandatory keys that are present.
+
+A ServiceForm RR is considered "compatible" with a client if the client
+implements support for all its mandatory keys.  If the SVCB RRSet contains
+no compatible RRs, the client will generally act as if the RRSet is empty.
+
+In presentation format, "mandatory" contains a list of one or more valid
+SvcParamKeys, either by their registered name or in the unknown-key format
+({{svcfieldvalue}}).  Keys MAY appear in any order, but MUST NOT appear more
+than once.  Any listed keys MUST also appear in the SvcFieldValue.
+For example, the following is a valid SvcFieldValue:
+
```

Comment:
Are "automatically mandatory" ones allowed to appear?  Is this "MUST NOT appear more than once" needed as a contraint for Presentation Format?
(May "automatically mandatory" entries appear in the wire format list?)

Reply:
> Are "automatically mandatory" ones allowed to appear?

I updated the text to make this a "SHOULD NOT".

> Is this "MUST NOT appear more than once" needed as a contraint for Presentation Format?

No, but there's no use in having duplicates, and it's easy for the zone file parser to detect.